test/src/unit-max-tile-size.cc

```diff
@@ -0,0 +1,1098 @@
+    // when all validity values are zero. dim_data should always contain data.
+    attr_data.reserve(dim_data.size());
+
+    // Prepare offsets to attr items...
+    std::vector<uint64_t> attr_offsets(dim_data.size(), 0);
+    // ...offsets for those for which actually have data;
+    for (decltype(attr_data.size()) i = 0; i < attr_data.size(); i++) {
+      attr_offsets[i] = i;
+    }
+    // ...pad out offsets for those which do not have data.
+    for (decltype(attr_data.size()) i = attr_data.size(); i < dim_data.size();
+         ++i) {
+      attr_offsets[i] = attr_data.size();
+    }
+
+    // Prepare the validity values for the data
+    std::vector<uint8_t> attr_val(dim_data.size());
+    // attr_val also needs a non empty buffer, and ...
+    attr_val.reserve(dim_data.size());
+    // ... init those for which there is data
+    for (decltype(attr_data.size()) i = 0; i < attr_data.size(); ++i) {
+      attr_val[i] = 1;
+    }
+    // ... init those trailing which are empty and have no data
+    for (decltype(attr_data.size()) i = attr_data.size(); i < dim_data.size();
+         ++i) {
+      attr_val[i] = 0;
+    }
+
+    // Open the array for writing and create the query.
+    tiledb::Array array(ctx_, main_array_name_, TILEDB_WRITE);
+    tiledb::Query query(ctx_, array);
+    query.set_layout(TILEDB_UNORDERED)
+        .set_data_buffer("a1", attr_data)
+        .set_offsets_buffer("a1", attr_offsets)
+        .set_validity_buffer("a1", attr_val)
+        .set_data_buffer("d1", dim_data)
+        .set_offsets_buffer("d1", dim_offsets);
+
+    // Perform the write and close the array.
+    query.submit();
+    array.close();
+  }
+
+  uint64_t c_get_fragments_max_in_memory_tile_size(
+      const std::string& array_uri) {
+    tiledb::Array array(ctx_, array_uri, TILEDB_READ);
+    uint64_t max_in_memory_tile_size = 0;
+    CHECK(
+        tiledb_array_maximum_tile_size(
+            &*array.ptr(), &max_in_memory_tile_size) == TILEDB_OK);
+
+    return max_in_memory_tile_size;
+  }
+
+  uint64_t cpp_get_fragments_max_in_memory_tile_size(
+      const std::string& array_uri) {
+    tiledb::Array array(ctx_, array_uri, TILEDB_READ);
+    uint64_t max_in_memory_tile_size = 0;
+    max_in_memory_tile_size = array.get_max_in_memory_tile_size();
+
+    return max_in_memory_tile_size;
+  }
+
+  tiledb::Context ctx_;
+  tiledb::VFS vfs_;
+};  // struct MaxTileSizeFx
+
+TEST_CASE_METHOD(
+    MaxTileSizeFx,
+    "C++ API: Max tile size, dense, fixed, with consolidation",
+    "[capi][cppapi][max-tile-size][dense][consolidate][vacuum]") {
+  tiledb_layout_t tile_order = TILEDB_ROW_MAJOR;
+  tiledb_layout_t cell_order = TILEDB_COL_MAJOR;
+  SECTION(" - tile/cell row major") {
+    tile_order = TILEDB_ROW_MAJOR;
+    cell_order = TILEDB_ROW_MAJOR;
+  }
+  SECTION(" - tile/cell col major") {
+    tile_order = TILEDB_COL_MAJOR;
+    cell_order = TILEDB_COL_MAJOR;
+  }
+  uint64_t capi_max;
+  uint64_t cppapi_max;
+
+  // Create array
```

Comment:
Replace with: // Create array and validate 0 max tile size.

Reply:
changed